btree/btree.c

```diff
@@ -8562,6 +8578,61 @@ M0_INTERNAL int64_t m0_btree_lrulist_purge(int64_t size)
 	m0_rwlock_write_unlock(&list_lock);
 	return total_size;
 }
+
+/**
+ * Checks whether lru list purging is required based on used space
+ * watermark(enum m0_btree_used_space_watermark).
+ *
+ * @param user purge user
+ * @param size requested memory to be freed.
+ *
+ * @return int total freed memory.
+ */
+M0_INTERNAL int64_t m0_btree_lrulist_purge_check(enum m0_btree_purge_user user,
+						 int64_t size)
+{
+	int64_t size_to_purge;
+	int64_t purged_size = 0;
+
+	if (lru_space_used < M0_USW_LOW) {
+		/** Do nothing. */
+		if (user == M0_PU_EXTERNAL)
+			M0_LOG(M0_INFO, "Used space is below threshold"
```

Comment:
In the string indicate something like "Skipping memory release since Used space is below .....". This way it makes sense to non-btree code reader or sysadmin that memory is not released for a reason.

Reply:
Done.